### PR TITLE
Infrastructure context

### PR DESCRIPTION
**Pull request by CDimonaco**
# Description

Create infrastructure context, cross-app concerns that are not related to a particular phoenix context.

## How was this tested?

Moved tests

### CODE REVIEW

File: lib/trento/event_handlers_supervisor.ex [hunk 1]
```diff
@@ -10,9 +10,9 @@ defmodule Trento.EventHandlersSupervisor do
   @impl true
   def init(_init_arg) do
     children = [
-      Trento.AlertsEventHandler,
-      Trento.RollUpEventHandler,
-      Trento.StreamRollUpEventHandler
+      Trento.Infrastructure.Commanded.EventHandlers.AlertsEventHandler,
```
Comment (arbulu89) [inline lib/trento/event_handlers_supervisor.ex:13]: We could alias these to make them shorter

File: lib/trento/infrastructure/commanded/roll_up/roll_up.ex [hunk 2]
```diff
@@ -1,12 +1,12 @@
-defmodule Trento.RollUp do
+defmodule Trento.Infrastructure.Commanded.RollUp do
```
Comment (arbulu89) [inline lib/trento/infrastructure/commanded/roll_up/roll_up.ex:1]: Following the rules, this should be `Commanded.RollUp.RollUp`, but I understand using it once looks better

File: lib/trento/infrastructure/discovery/payloads/cloud_discovery_payload.ex [hunk 3]
```diff
@@ -1,4 +1,4 @@
-defmodule Trento.Integration.Discovery.CloudDiscoveryPayload do
+defmodule Trento.Infrastructure.Discovery.CloudDiscoveryPayload do
```
Comment (arbulu89) [inline lib/trento/infrastructure/discovery/payloads/cloud_discovery_payload.ex:1]: `Infrastructure.Discovery.Payloads.CloudDiscoveryPayload`
The same for the other payloads

File: lib/trento/infrastructure/discovery/policies/cluster_policy.ex [hunk 4]
```diff
@@ -1,4 +1,4 @@
-defmodule Trento.Integration.Discovery.ClusterPolicy do
+defmodule Trento.Infrastructure.Discovery.ClusterPolicy do
```
Comment (arbulu89) [inline lib/trento/infrastructure/discovery/policies/cluster_policy.ex:1]: `Infrastructure.Discovery.Policies.ClusterPolicy`

File: lib/trento/infrastructure/installation/installation.ex [hunk 5]
```diff
@@ -1,4 +1,4 @@
-defmodule Trento.Installation do
+defmodule Trento.Infrastructure.Installation do
```
Comment (arbulu89) [inline lib/trento/infrastructure/installation/installation.ex:1]: Maybe this should even be a `context` itself, what do you think?

File: lib/trento/support/mix/tasks/dump_scenario.ex [hunk 6]
```diff
@@ -5,9 +5,9 @@ defmodule Mix.Tasks.DumpScenario do
   use Mix.Task
   import Trento.Tasks.Helper
 
-  alias Trento.Integration.Discovery
+  alias Trento.Infrastructure.Discovery
```
Comment (arbulu89) [inline lib/trento/support/mix/tasks/dump_scenario.ex:8]: Not this line specifically, but this module shouldn't be `Mix.Support.Tasks.DumpScenario`.
In general, do we omit the `Support` folder namespace?

File: mix.exs [hunk 7]
```diff
@@ -172,8 +172,8 @@ defmodule Trento.MixProject do
       nest_modules_by_prefix: [
         Trento.Domain.Events,
         Trento.Domain.Commands,
-        Trento.Integration.Checks,
-        Trento.Integration.Discovery
+        Trento.Infrastructure.Checks,
```
Comment (arbulu89) [inline mix.exs:175]: Mental note, we need to change the 2 lines above once we finish the migration of all events

File: test/trento/infrastructure/commanded/roll_up/roll_up_test.exs [hunk 8]
```diff
@@ -1,7 +1,7 @@
-defmodule Trento.RollUpTest do
+defmodule Trento.Infrastructure.Commanded.RollUp.RollUpTest do
```
Comment (arbulu89) [inline test/trento/infrastructure/commanded/roll_up/roll_up_test.exs:1]: Shouldn't this be `Commanded.RollUpTest` simply, unless you change the lib code

**Review by arbulu89 — COMMENTED** [review]
Hey @CDimonaco 
Noice!
Great job.
I just noted some comments

File: lib/trento/infrastructure/installation/installation.ex (hunk 5 above)
Comment (CDimonaco) [reply lib/trento/infrastructure/installation/installation.ex:1]: I was thinking about that, the same mixed feeling with alerting, I don't know if it "qualifies" as a phoenix context, but yes I have the same feeling, let's investigate further

File: lib/trento/infrastructure/discovery/payloads/cloud_discovery_payload.ex (hunk 3 above)
Comment (CDimonaco) [reply lib/trento/infrastructure/discovery/payloads/cloud_discovery_payload.ex:1]: done

File: lib/trento/infrastructure/discovery/policies/cluster_policy.ex (hunk 4 above)
Comment (CDimonaco) [reply lib/trento/infrastructure/discovery/policies/cluster_policy.ex:1]: done

File: test/trento/infrastructure/commanded/roll_up/roll_up_test.exs (hunk 8 above)
Comment (CDimonaco) [reply test/trento/infrastructure/commanded/roll_up/roll_up_test.exs:1]: The lib code is `Trento.Infrastructure.Commanded.RollUp`

Comment (CDimonaco) [reply test/trento/infrastructure/commanded/roll_up/roll_up_test.exs:1]: yep, thx

File: lib/trento/event_handlers_supervisor.ex (hunk 1 above)
Comment (CDimonaco) [reply lib/trento/event_handlers_supervisor.ex]: done

File: lib/trento/infrastructure/commanded/roll_up/roll_up.ex (hunk 2 above)
Comment (CDimonaco) [reply lib/trento/infrastructure/commanded/roll_up/roll_up.ex:1]: Yeah it's true, it just feels better but I think we need to adhere to the convention, having this exceptions in the beginning could lead to bad example/direction in the long term, wdyt?

File: lib/trento/support/mix/tasks/dump_scenario.ex (hunk 6 above)
Comment (CDimonaco) [reply lib/trento/support/mix/tasks/dump_scenario.ex:8]: I added the namespace, thx for spotting it

File: lib/trento/support/ecto/payload.ex [hunk 9]
```diff
@@ -1,4 +1,4 @@
-defmodule Ecto.Payload do
+defmodule Ecto.Support.Payload do
```
Comment (arbulu89) [inline lib/trento/support/ecto/payload.ex:1]: Maybe this should be `Trento.Support.Payload`

File: lib/trento/infrastructure/commanded/roll_up/roll_up.ex (hunk 2 above)
Comment (arbulu89) [reply lib/trento/infrastructure/commanded/roll_up/roll_up.ex:1]: So, let's rename it to `RoullUp.RollUp` them

File: lib/trento/infrastructure/discovery/payloads/cluster/cluster_discovery_payload.ex [hunk 10]
```diff
@@ -1,4 +1,4 @@
-defmodule Trento.Integration.Discovery.ClusterDiscoveryPayload do
+defmodule Trento.Infrastructure.Discovery.ClusterDiscoveryPayload do
```
Comment (arbulu89) [inline lib/trento/infrastructure/discovery/payloads/cluster/cluster_discovery_payload.ex:1]: Shouldn't this be `Discovery.Payloads.Cluster.ClusterDiscoveryPayload` ?

File: lib/trento/infrastructure/discovery/payloads/cluster/cib_discovery_payload.ex [hunk 11]
```diff
@@ -1,4 +1,4 @@
-defmodule Trento.Integration.Discovery.ClusterDiscoveryPayload.Cib do
+defmodule Trento.Infrastructure.Discovery.Payloads.Cluster.ClusterDiscoveryPayload.Cib do
```
Comment (arbulu89) [inline lib/trento/infrastructure/discovery/payloads/cluster/cib_discovery_payload.ex:1]: `Payloads.Cluster.CibDiscoveryPayload` or `Payloads.Cluster.Cib`

File: lib/trento/infrastructure/discovery/payloads/cluster/crmmon_discovery_payload.ex [hunk 12]
```diff
@@ -1,4 +1,4 @@
-defmodule Trento.Integration.Discovery.ClusterDiscoveryPayload.Crmmon do
+defmodule Trento.Infrastructure.Discovery.Payloads.Cluster.ClusterDiscoveryPayload.Crmmon do
```
Comment (arbulu89) [inline lib/trento/infrastructure/discovery/payloads/cluster/crmmon_discovery_payload.ex:1]: `Payloads.Clusters.Crmmon`

File: lib/trento/infrastructure/discovery/payloads/cluster/sbd_discovery_payload.ex [hunk 13]
```diff
@@ -1,11 +1,11 @@
-defmodule Trento.Integration.Discovery.ClusterDiscoveryPayload.Sbd do
+defmodule Trento.Infrastructure.Discovery.Payloads.Cluster.ClusterDiscoveryPayload.Sbd do
```
Comment (arbulu89) [inline lib/trento/infrastructure/discovery/payloads/cluster/sbd_discovery_payload.ex:1]: `Payloads.Cluster.Sbd`

File: test/trento/infrastructure/discovery/policies/cluster_policy_test.exs [hunk 14]
```diff
@@ -1,14 +1,14 @@
-defmodule Trento.Integration.Discovery.ClusterPolicyTest do
+defmodule Trento.Infrastructure.Discovery.ClusterPolicyTest do
```
Comment (arbulu89) [inline test/trento/infrastructure/discovery/policies/cluster_policy_test.exs:1]: `Discovery.Policies.ClusterPolicyTest`

File: test/trento/infrastructure/discovery/policies/host_policy_test.exs [hunk 15]
```diff
@@ -1,11 +1,11 @@
-defmodule Trento.Integration.Discovery.HostPolicyTest do
+defmodule Trento.Infrastructure.Discovery.HostPolicyTest do
```
Comment (arbulu89) [inline test/trento/infrastructure/discovery/policies/host_policy_test.exs:1]: `Discovery.Policies.HostPolicyTest`

File: test/trento/infrastructure/discovery/policies/sap_system_policy_test.exs [hunk 16]
```diff
@@ -1,14 +1,14 @@
-defmodule Trento.Integration.Discovery.SapSystemPolicyTest do
+defmodule Trento.Infrastructure.Discovery.SapSystemPolicyTest do
```
Comment (arbulu89) [inline test/trento/infrastructure/discovery/policies/sap_system_policy_test.exs:1]: `Discovery.Policies.SapSystemPolicyTest`

**Review by arbulu89 — CHANGES_REQUESTED** [review]
Hey @CDimonaco 
Some more namespace changes XD

File: CHANGELOG.md [hunk 17]
```diff
@@ -1642,7 +1642,7 @@
 - Close filters when clicking outside [\#260](https://github.com/trento-project/web/pull/260) ([fabriziosestito](https://github.com/fabriziosestito))
 - Add checks execution domain logic [\#256](https://github.com/trento-project/web/pull/256) ([fabriziosestito](https://github.com/fabriziosestito))
 - Cluster details with IPs and filters enabled app-wide [\#254](https://github.com/trento-project/web/pull/254) ([dottorblaster](https://github.com/dottorblaster))
-- Fix Trento.Type error handling and adds tests [\#252](https://github.com/trento-project/web/pull/252) ([fabriziosestito](https://github.com/fabriziosestito))
+- Fix Trento.Support.Type error handling and adds tests [\#252](https://github.com/trento-project/web/pull/252) ([fabriziosestito](https://github.com/fabriziosestito))
```
[the remark this answers is not in the document]
Comment (CDimonaco) [reply CHANGELOG.md:1645]: done

File: lib/trento/support/ecto/payload.ex (hunk 9 above)
Comment (CDimonaco) [reply lib/trento/support/ecto/payload.ex:1]: done

**Review by arbulu89 — APPROVED** [review]
@CDimonaco 
All good!
Green light from my side

**Pull request merged**